### PR TITLE
Task vv/19 Added modal component

### PR DESCRIPTION
**Pull request by vitalik-prog**

### CODE REVIEW

File: frontend/src/components/common/modal/modal.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <ReactModal
+      isOpen={showModal}
+      shouldCloseOnOverlayClick={true}
+      preventScroll={false}
+      className={styles.modal}
+      overlayClassName={styles.overlay}
+      appElement={document.getElementById('root') as HTMLElement}
+    >
+      {children}
+      <button className={styles.closeBtn} onClick={handleCloseModal}> X </button>
```
Comment (what1s1ove) [inline frontend/src/components/common/modal/modal.tsx]:
```suggestion
```

File: frontend/src/components/common/modal/modal.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const Modal: React.FC<Props> = ({
+  showModal = true,
+  handleCloseModal,
+  children,
+}) => {
+
+  return (
+    <ReactModal
+      isOpen={showModal}
+      shouldCloseOnOverlayClick={true}
```
Comment (what1s1ove) [inline frontend/src/components/common/modal/modal.tsx]:
```suggestion
      shouldCloseOnOverlayClick
```

File: frontend/src/components/common/modal/modal.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import ReactModal from 'react-modal';
+import styles from './styles.module.scss';
+
+type Props = {
+  showModal: boolean;
```
Comment (what1s1ove) [inline frontend/src/components/common/modal/modal.tsx]:
```suggestion
  isShowModal: boolean;
```

File: frontend/src/components/common/modal/modal.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  children: React.ReactNode;
```
Comment (what1s1ove) [inline frontend/src/components/common/modal/modal.tsx]:
```suggestion
```

File: frontend/package.json [hunk 5]
```diff
@@ -17,6 +17,7 @@
     "@hookform/error-message": "2.0.0",
     "@hookform/resolvers": "2.6.1",
     "@reduxjs/toolkit": "1.6.0",
+    "@types/react-modal": "^3.12.1",
```
Comment (what1s1ove) [inline frontend/package.json]: check **B17**

File: frontend/src/components/common/modal/modal.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  handleCloseModal?: () => void;
```
Comment (what1s1ove) [inline frontend/src/components/common/modal/modal.tsx]: check **A8**

File: package-lock.json [hunk 7]
```diff
@@ -1,3402 +1,3402 @@
-{
```
Comment (what1s1ove) [inline package-lock.json]: it should be changed

**Review by what1s1ove — CHANGES_REQUESTED** [review]
frontend/package-lock.json shouldn't be changed so much

File: frontend/package.json [hunk 8]
```diff
@@ -17,6 +17,7 @@
     "@hookform/error-message": "2.0.0",
     "@hookform/resolvers": "2.6.1",
     "@reduxjs/toolkit": "1.6.0",
+    "@types/react-modal": "3.12.1",
```
Comment (what1s1ove) [inline frontend/package.json]: types should be in the `devDependencies`

File: package-lock.json (hunk 7 above)
Comment (what1s1ove) [reply package-lock.json]: don't fixed

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by JuliaDzyuba — DISMISSED** [review]

File: package-lock.json (hunk 7 above)
Comment (what1s1ove) [reply package-lock.json]: don't fixed

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**